agents/ansari_workflow.py

```diff
@@ -0,0 +1,193 @@
+import hashlib
+import logging
+import os
+import sys
+from datetime import date
+
+import litellm
+
+from tools.search_hadith import SearchHadith
+from tools.search_vectara import SearchVectara
+from tools.search_quran import SearchQuran
+from util.prompt_mgr import PromptMgr
+
+logger = logging.getLogger(__name__ + ".AnsariWorkflow")
+
+if not sys.argv[0].endswith("main_api.py"):
+    logging_mode = logging.DEBUG
+else:
+    logging_mode = logging.INFO
+
+logger.setLevel(logging_mode)
+
+# # Uncomment below when logging above doesn't output to std, and you want to see the logs in the console
+# console_handler = logging.StreamHandler()
+# console_handler.setLevel(logging_mode)
+# logger.addHandler(console_handler)
+
+
+class AnsariWorkflow:
+    """
+    AnsariWorkflow manages the execution of modular workflow steps for processing user queries.
```

Comment:
This seems to overlap a little bit with LangChain, but let's try it for now.

Reply:
Yes :) I hadn't considered LangChain initially, probably because I perceived it as overly complex. However, you make a good point - LangChain offers numerous valuable features. If we need that functionality, switching to LangChain would make sense. LangGraph is also quite impressive.